app/fetchfile.py

```diff
@@ -0,0 +1,42 @@
+import os
+
+import docx
+import requests
+from flask import send_file, send_from_directory
+
+
+class Fetchfile(object):
+
+    def __init__(self, app=None):
+        self.app = app
+        self._linker_url = None
+        self._base_dir = None
+
+        if app is not None:
+            self.init_app(app)
+
+    def init_app(self, app):
+        self._base_dir = self.app.config['BASE_DIR']
+        self._linker_url = self.app.config['LINKER_URL']
+        app.fetchfile = self
+
+    def fetch_doc(self, uid, outdir):
+        if not self.file_is_doc(uid):
+            return Exception
+        url = self._linker_url + uid
+        r = requests.get(url, allow_redirects=True)
+        name = r.url.split('/')[-1]
+        out_path = os.path.join(outdir, name)
```

Comment:
This should be with cache semantics. Meaning, we shouldn't download the file if we already have it.
Something like we do [here](https://github.com/Bnei-Baruch/archive-unzip/blob/fdf447c431a68c18c7c4c659891c335a46ae9943/app/doc2html/views.py#L139)

Use sha1 for ensuring it's the same file content. file names are not robust enough for that.